src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -5922,12 +5932,24 @@ private void ParseTypeArgumentList(out SyntaxToken open, SeparatedSyntaxListBuil
             // remaining types & commas
             while (true)
             {
-                if (this.CurrentToken.Kind == SyntaxKind.GreaterThanToken)
+                // Those token kinds are our terminator token kinds that break type argument parsing
+                // It is more useful to report fewer redundant and cryptic errors than to hint using
+                // tuples as possible types
```

Comment:
This comment is a bit cryptic and hard to read. The first sentence seems somewhat unnecessary: the function name conveys that info. I think it would be better talk about the reason for this handling, something like `// We prefer parsing as if there was a missing > over parsing as a tuple type for error recovery`.